source/github-helpers/github-url.ts

```diff
@@ -39,14 +11,49 @@ export default class GitHubURL extends URL {
 	branch: string;
 	// @ts-ignore
 	filePath: string;
+
+	assign = Object.assign.bind(null, this);
+
 	constructor(url: string) {
 		super(url);
 		this.pathname = super.pathname;
 	}
 
-	assign(replacements: Partial<GitHubURL>): this {
-		Object.assign(this, replacements);
-		return this;
+	toString() {
+		return this.href;
+	}
+
+	private disambiguateReference(ambiguousReference: string[]): {branch: string; filePath: string} {
+		const branch = ambiguousReference[0];
+		const filePathFromSearch = this.searchParams.getAll('path[]').join('/');
```

Comment:
Needs a comment explaining when this happens. e.g. `// History pages might use search parameters`

Reply:
Will add